python/storage/README.md

```diff
@@ -0,0 +1,145 @@
+# kserve-storage
+
+A Python module for handling model storage and retrieval for KServe. This package provides a unified API to download models from various storage backends including cloud providers, file systems, and model hubs.
+
+## Features
+
+- Support for multiple storage backends:
+    - Local file system
+    - Google Cloud Storage (GCS)
+    - Amazon S3
+    - Azure Blob Storage
+    - Azure File Share
+    - HTTP/HTTPS URLs
+    - HDFS/WebHDFS
+    - Hugging Face Hub
+- Automatic extraction of compressed files (zip, tar.gz, tgz)
+- Configuration via environment variables
+- Logging and error handling
+
+## Installation
+
+```bash
+pip install kserve-storage
+```
+
+Or with Poetry:
+
+```bash
+poetry add kserve-storage
+```
+
+## Usage
+
+The main entry point is the `Storage` class which provides a `download` method:
+
+```python
+from kserve_storage import Storage
+
+# Download from GCS to a temporary directory
+model_dir = Storage.download("gs://your-bucket/model")
+
+# Download from S3 to a specific directory
+model_dir = Storage.download("s3://your-bucket/model", "/path/to/destination")
+```
+
+## Supported Storage Providers
+
+### Local File System
+
+```python
+model_dir = Storage.download("file:///path/to/model")
+# or using direct path
+model_dir = Storage.download("/path/to/model")
+```
+
+### Google Cloud Storage
+
+```python
+model_dir = Storage.download("gs://bucket-name/model-path")
+```
+
+### Amazon S3
+
+```python
+model_dir = Storage.download("s3://bucket-name/model-path")
+```
+
+### Azure Blob Storage
+
+```python
+model_dir = Storage.download("https://account-name.blob.core.windows.net/container-name/model-path")
+```
+
+### Azure File Share
+
+```python
+model_dir = Storage.download("https://account-name.file.core.windows.net/share-name/model-path")
+```
+
+### HTTP/HTTPS URLs
+
+```python
+model_dir = Storage.download("https://example.com/path/to/model.zip")
+```
+
+### HDFS
+
+```python
+model_dir = Storage.download("hdfs://path/to/model")
+# or WebHDFS
+model_dir = Storage.download("webhdfs://path/to/model")
+```
+
+### Hugging Face Hub
+
+```python
+model_dir = Storage.download("hf://org-name/model-name")
+# With specific revision
+model_dir = Storage.download("hf://org-name/model-name:revision")
+```
+
+## Environment Variables
```

Comment:
How to set HF token to pull a model from Huggingface hub private repo?

Reply:
Yes I know how to set the token but my question was, we don't need to add the part in this doc like others or not

Reply:
Maybe just link the huggingface docs, what do you think?